pkg/doc/verifiable/example_presentation_test.go

```diff
@@ -163,3 +167,290 @@ func ExampleCredential_Presentation() {
+	}
+
+	//Output:
+}
+
+func ExamplePresentation_MarshalJSON() {
+	vp := &verifiable.Presentation{
+		Context: []string{
+			"https://www.w3.org/2018/credentials/v1"},
+		ID:     "urn:uuid:3978344f-8596-4c3a-a978-8fcaba3903c",
+		Type:   []string{"VerifiablePresentation"},
+		Holder: "did:example:ebfeb1f712ebc6f1c276e12ec21",
+	}
+
+	vc := verifiable.Credential{
+		Context: []string{
+			"https://www.w3.org/2018/credentials/v1",
+			"https://www.w3.org/2018/credentials/examples/v1"},
+		ID: "http://example.edu/credentials/1872",
+		Types: []string{
+			"VerifiableCredential",
+			"UniversityDegreeCredential"},
+		Subject: UniversityDegreeSubject{
+			ID:     "did:example:ebfeb1f712ebc6f1c276e12ec21",
+			Name:   "Jayden Doe",
+			Spouse: "did:example:c276e12ec21ebfeb1f712ebc6f1",
+			Degree: UniversityDegree{
+				Type:       "BachelorDegree",
+				University: "MIT",
+			},
+		},
+		Issuer: verifiable.Issuer{
+			ID:   "did:example:76e12ec712ebc6f1c221ebfeb1f",
+			Name: "Example University",
+		},
+		Issued:  &issued,
+		Expired: &expired,
+		Schemas: []verifiable.TypedID{},
+		CustomFields: map[string]interface{}{
+			"referenceNumber": 83294847,
+		},
+	}
+
+	err := vp.SetCredentials(vc)
+	if err != nil {
+		fmt.Println(fmt.Errorf("failed to set credentials of VP: %w", err))
+	}
+
+	// json.MarshalIndent() calls Presentation.MarshalJSON()
+	vpJSON, err := json.MarshalIndent(vp, "", "\t")
+	if err != nil {
+		fmt.Println(fmt.Errorf("failed to marshal VP to JSON: %w", err))
+	}
+
+	fmt.Println(string(vpJSON))
+
+	// Output: {
+	//	"@context": [
+	//		"https://www.w3.org/2018/credentials/v1"
+	//	],
+	//	"id": "urn:uuid:3978344f-8596-4c3a-a978-8fcaba3903c",
+	//	"type": [
+	//		"VerifiablePresentation"
+	//	],
+	//	"verifiableCredential": [
+	//		{
+	//			"Context": [
+	//				"https://www.w3.org/2018/credentials/v1",
+	//				"https://www.w3.org/2018/credentials/examples/v1"
+	//			],
+	//			"CustomContext": null,
+	//			"ID": "http://example.edu/credentials/1872",
+	//			"Types": [
+	//				"VerifiableCredential",
+	//				"UniversityDegreeCredential"
+	//			],
+	//			"Subject": {
+	//				"id": "did:example:ebfeb1f712ebc6f1c276e12ec21",
+	//				"name": "Jayden Doe",
+	//				"spouse": "did:example:c276e12ec21ebfeb1f712ebc6f1",
+	//				"degree": {
+	//					"type": "BachelorDegree",
+	//					"university": "MIT"
+	//				}
+	//			},
+	//			"Issuer": {
+	//				"ID": "did:example:76e12ec712ebc6f1c221ebfeb1f",
+	//				"Name": "Example University"
+	//			},
+	//			"Issued": "2010-01-01T19:23:24Z",
+	//			"Expired": "2020-01-01T19:23:24Z",
+	//			"Proof": null,
+	//			"Status": null,
+	//			"Schemas": [],
+	//			"Evidence": null,
+	//			"TermsOfUse": null,
+	//			"RefreshService": null,
+	//			"CustomFields": {
+	//				"referenceNumber": 83294847
+	//			}
+	//		}
+	//	],
+	//	"holder": "did:example:ebfeb1f712ebc6f1c276e12ec21"
+	//}
+}
+
+//nolint:gocyclo
+func ExamplePresentation_MarshalledCredentials() {
```

Comment:
Added example of Verifiable.MarshalledCredentials(). In this example, it's shown:
- Holder prepares VP in JWS form where VP is defined in JWS form as well.
- Verifier decodes VP and then decodes its VC.